cdk/app/local_office_search_api.py

```diff
@@ -0,0 +1,122 @@
+from aws_cdk import Stack, Stage
+from aws_cdk.aws_ecr import Repository
+from aws_cdk.aws_ec2 import Vpc
+from aws_cdk.aws_eks import ServiceAccount
+from aws_cdk.aws_rds import Credentials, DatabaseCluster
+from aws_cdk.aws_s3 import Bucket
+from aws_cdk.aws_secretsmanager import Secret, SecretStringGenerator
+from ca_cdk_constructs.eks.external_secrets import (
+    ExternalAwsSecretsChart,
+    ExternalSecretSource,
+)
+from constructs import Construct
+
+from cdk8s import App
+from ca_cdk_constructs.eks.eks_cluster_integration import EksClusterIntegration
+
+from .chart import LocalOfficeSearchApiChart
+
+
+class LocalOfficeSearchApiDeployment(Stack):
+    def __init__(self,
+                 scope: Construct,
+                 construct_id: str,
+                 app_image_version: str,
+                 db: DatabaseCluster,
+                 db_credentials: Credentials,
+                 lss_bucket_name: str,
+                 geo_data_bucket_name: str,
+                 geo_data_postcode_file: str,
+                 **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        image_repo = Repository.from_repository_arn(self, "LocalOfficeSearchApiRepo", "arn:aws:ecr:eu-west-1:979633842206:repository/local-office-search-api")
+
+        vpc = Vpc.from_lookup(
+            self,
+            "VpcLookup",
+            is_default=False,
+            region="eu-west-1",
+            tags={"Product": "vpc"},
+        )
+
+        namespace = f"{Stage.of(self).stage_name}-local-office-search-api"
+
+        eks_cluster = EksClusterIntegration(
+            self,
+            "EksClusterIntegration",
+            vpc=vpc,
+            cluster_name=f"{Stage.of(self).stage_name}-eks-platform",
+            role_name="LocalOfficeSearchApiDeploymentEksClusterIntegration",
+        ).cluster
+
+        lss_data_bucket = Bucket.from_bucket_name(self, "LssBucket", bucket_name=lss_bucket_name)
+        geo_data_bucket = Bucket.from_bucket_name(self, "GeoDataBucket", bucket_name=geo_data_bucket_name)
+
+        app_secrets_secret = Secret(
+            self,
+            "AppSecrets",
+            secret_name=f"content-platform-LocalOfficeSearchApiAppSecrets-{Stage.of(self).stage_name}",
+            generate_secret_string=SecretStringGenerator()
+        )
+
+        self._service_account = ServiceAccount(
+            self,
+            "LocalOfficeSearchApiServiceAccount",
+            namespace=namespace,
+            name="local-office-search-api",
+            cluster=eks_cluster,
+            labels={"app.kubernetes.io/name": "local-office-search-api"},
+        )
+        lss_data_bucket.grant_read(self._service_account)
+        geo_data_bucket.grant_read(self._service_account)
+
+        self._cdk8s_app = App()
+
+        rds_secret_source = ExternalSecretSource(
+            source_secret=db_credentials.secret_name,
+            secret_mappings={"password": "DB_PASSWORD"},
```

Comment:
why only the password, the app would need the endpoints and the username too, right?

Reply:
I've not considered those to be secrets so I'm passing them directly through from the DB infra stack and then using `from_value` https://github.com/citizensadvice/local-office-search-api/blob/34872647b8e977dd4eb2f11cc1676a166d7a6b8d/cdk/app/chart.py#L167-L170

Reply:
but then you're not reading the credentials from the k8s secret, including the password, you're injecting them directly, it doesn't seem right, you should be using `envFrom` to get it from the secret... I would put all the db credentials together to make it simpler but that's up to you

Reply:
or maybe cdk8s does that for you? I'm not familiar with the syntax

Reply:
ah it's a cdk8s+ thing ...